pkg/agent/core/ngt/service/ngt.go

```diff
@@ -198,26 +210,96 @@ func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 	return n, nil
 }
 
+// migrate migrates the index directory from old to new under the input path if necessary.
+// Migration happens when the path is not empty and there is no `path/origin` directory,
+// which indicates that the user has NOT been using CoW mode and the index directory is not migrated yet.
+func migrate(path string) (err error) {
+	// check if migration is required
+	files, err := file.ListInDir(path)
+	if err != nil {
+		return err
+	}
+	if len(files) == 0 {
+		// empty directory doesn't need migration
+		return nil
+	}
+
+	originExists := false
+	for _, file := range files {
+		if file == path+string(os.PathSeparator)+string(originIndexDirName) {
```

Comment:
Is it better to use file.Join to join the path?

```suggestion
		if file == file.Join(path, string(originIndexDirName)) {
```

Reply:
👍 
[8354446](https://github.com/vdaas/vald/pull/2034/commits/83544469b708c42bd9018a5136677bc8c5d52ff9)